src/Development/IDE/Types/Location.hs

```diff
@@ -49,22 +49,55 @@ import Language.Haskell.LSP.Types as LSP (
   )
 import SrcLoc as GHC
 import Text.ParserCombinators.ReadP as ReadP
+import GHC.Generics
 
 
 -- | Newtype wrapper around FilePath that always has normalized slashes.
-newtype NormalizedFilePath = NormalizedFilePath FilePath
-    deriving (Eq, Ord, Show, Hashable, NFData, Binary)
+-- The NormalizedUri and hash of the FilePath are cached to avoided
+-- repeated normalisation when we need to compute them (which is a lot).
+--
+-- This is one of the most performance critical parts of ghcide, do not
+-- modify it without profiling.
+data NormalizedFilePath = NormalizedFilePath NormalizedUriWrapper Int !FilePath
```

Comment:
Similar question as for the `haskell-lsp` PR. Should we make the first two fields or at least the hash strict and unpack it?

Reply:
I can see the argument for making the hash strict but not the `NormalizedUriWrapper` as that is quite expensive to compute if you never use it.

Reply:
I made the hash strict.

Reply:
Thanks!